src/popover/popover.js

```diff
@@ -418,30 +419,35 @@ class Popover extends React.Component<PopoverPropsT, PopoverPrivateStateT> {
             {...getOverrideProps(ArrowOverride)}
           />
         ) : null}
-        <Inner
-          key="popover-inner"
-          {...sharedProps}
-          {...getOverrideProps(InnerOverride)}
-        >
-          {typeof content === 'function' ? content() : content}
+        <Inner {...sharedProps} {...getOverrideProps(InnerOverride)}>
+          {renderedContent}
         </Inner>
       </Body>
     );
   }
 
+  renderContent() {
+    const {content} = this.props;
+    return typeof content === 'function' ? content() : content;
+  }
+
   render() {
+    const mountedAndOpen = this.state.isMounted && this.props.isOpen;
     const rendered = [this.renderAnchor()];
+    const renderedContent =
+      mountedAndOpen || this.props.renderAll ? this.renderContent() : null;
+
     const defaultPopperOptions = {
       modifiers: {
         preventOverflow: {enabled: !this.props.ignoreBoundary},
       },
     };
     // Only render popover on the browser (portals aren't supported server-side)
-    if (__BROWSER__) {
-      if (this.state.isMounted && this.props.isOpen) {
+    if (renderedContent) {
+      if (__BROWSER__ && mountedAndOpen) {
```

Comment:
I believe the `__BROWSER__` check in `Popover` is not required since we won't get `componentDidMount` called until it hits the client anyway. The browser check was in the initial implementation and looks like it was superseded in https://github.com/uber/baseweb/pull/977